include/manager.hpp

```diff
@@ -39,6 +40,117 @@ class Manager
      */
     ~Manager() = default;
 
+    /**
+     * @brief Update keyword value.
+     *
+     * API can be used to update VPD keyword.
+     *
+     * To update a keyword on cache, input path should be d-bus object path.
+     * Eg: ("/xyz/openbmc_project/inventory/system/chassis/motherboard").
+     *
+     * To update a keyword on hardware, input path should be EEPROM path.
+     * Eg:
+     * ("/sys/bus/i2c/drivers/at24/9-0052/eeprom").
+     *
+     * To update a keyword on both cache and hardware, input path should be
+     * d-bus object path. Eg:
+     * ("/xyz/openbmc_project/inventory/system/chassis/motherboard").
+     *
+     * To update IPZ type VPD, input data should be in the form of (Record,
+     * Keyword, Value). Eg: ("VINI", "SN", {0x01, 0x02, 0x03}).
+     *
+     * To update Keyword type VPD, input data should be in the form of (Keyword,
+     * Value). Eg: ("PE", {0x01, 0x02, 0x03}).
+     *
+     * The target to update can either be cache/hardware/both, whose values are
+     * 0/1/2 respectively.
+     *
+     * @param[in] i_path - D-bus object path/EEPROM path.
+     * @param[in] i_data - Data to be updated.
+     * @param[in] i_target - Target location to update (0/1/2).
+     *
+     * @return On failure this API throws an error, on success it returns
+     * nothing.
+     */
+    void updateKeyword(const types::Path i_path, const types::VpdData i_data,
+                       const uint8_t i_target);
+
+    /**
+     * @brief Read keyword value.
+     *
+     * API can be used to read VPD keyword.
+     *
+     * To read a keyword from cache, input path should be d-bus object path.
+     * Eg: ("/xyz/openbmc_project/inventory/system/chassis/motherboard").
+     *
+     * To read a keyword from hardware, input path should be EEPROM path.
+     * Eg: ("/sys/bus/i2c/drivers/at24/9-0052/eeprom").
+     *
+     * To read keyword from IPZ type VPD, input data should be in the form of
+     * (Record, Keyword). Eg: ("VINI", "SN").
+     *
+     * To read keyword from keyword type VPD, just keyword name has to be
+     * supplied as input. Eg: ("SN").
+     *
+     * The target to read can either be cache/hardware, whose values are 0/1
+     * respectively.
+     *
+     * @param[in] i_path - D-bus object path/EEPROM path.
+     * @param[in] i_data - Data to be read.
+     * @param[in] i_target - Target location to read from (0/1).
+     *
+     * @return Read value in array of bytes.
+     */
+    types::BinaryVector readKeyword(const types::Path i_path,
+                                    const types::VpdData i_data,
+                                    const uint8_t i_target);
+
+    /**
+     * @brief Collect single FRU VPD
+     * API can be used to perform VPD collection for the given FRU.
```

Comment:
For a FRU to get collected via this API it should have any flag set in the JSON? If yes, then mention that here in details.